### ISSUE TITLE
(Windows 10) persistent error from Vuze days; update constantly requested and fails due to 'missing signature.'

### ISSUE THREAD

**Issue by pootthedestroyer** [BiglySoftware/BiglyBT#1655]
System Info:

Java 1.8.0_202 (64 bit)
  Oracle Corporation
c:\program files\biglybt\jre

SWT v4930r7, win32, zoom=100, dpi=96
Windows 10 v10.0, amd64 (64 bit)
B2.3.0.0/4 az3

--------

Like Vuze before it, my BiglyBt build has fallen into an endless cycle of the following two prompt windows:

1) "Software Update: BiglyBt just completed downloading an important update and must now be restarted so that the update can be installed."

2) Install Verification Failed: "Verification of 'Core BiglyBt Version' failed: Signature missing from file."

This bug is maximally disruptive, as every time these prompts appear, they force the application to take precedence over any other application currently running... and there appears to be no way to solve the problem short of a total reinstall. Indeed, there's really no information provided whatsoever by the application as to what this 'important update' is, or what signature is missing from what file. Notably, unlike many situations where BiglyBt/Vuze updates itself, no files were automatically downloaded via torrent, so there's nothing in my download library to help narrow down the issue.

**Comment by parg** [comment]
Could it be caused by the update occurring twice? That is, you get one update prompt while away from the computer and before you return to it the updater kicks in again and produces a second prompt 'on top of' the previous one? I've seen this happen and it confused the update process - obviously the second update should be disabled until the first one has been dealt with.

**Comment by pootthedestroyer** [comment]
Well, I only get one of each popup at a time (though they occur simultaneously.) You would think there'd be some duplication were that the case... unless of course the double-update is an instigating event that throws the program into the endless loop. That's certainly possible.

The problem is, of course, that this never stops. Never. Only uninstall/reinstall fixes it, until of course it happens again (though it does take awhile, which lends credence to the hypothesis that there's a discrete instigating event.)

**Comment by parg** [comment]
Are there any clues in the 'update.log' file in %appdata%\BiglyBT?

Also, after you get the 'must restart' prompt is there an alert indicator bottom-right of the status bar? Clicking on that might tell you what has updated.

**Comment by pootthedestroyer** [comment]
None whatsoever, which is very frustrating. The update log simply has more entries than usual clustered up around a week ago to now, but not as many as I'd expect, and they're not any different at all from older entries still in the log file - literally identical, except for the date and maybe the maxmem count.

Additionally, no alerts down at the bottom right (I'm assuming you mean the place where the little blue 'i' and/or yellow triangle exclamation point warnings will appear.)

**Comment by parg** [comment]
In the Library detailed view do you have any 'update torrents' in a borked state? Given you are on 2.3.0.0 the only update that you should be getting is the 2.4.0.0 full installer (core_BiglyBT_Update_2.4.0.0_win64.zip)

**Comment by pootthedestroyer** [comment]
There's nothing in my library right now except previously downloaded non-update-related torrents;  However, once I read your comment I did go ahead and update manually to 2.4.0.0. To the best of my recollection I have never seen the aforementioned update downloaded, or even attempted. Maybe that's where the problem lay.

Since I manually updated, predictably, the problem has not reappeared. I sadly must predict that it will recur eventually - possibly the next time there's some mandatory update.
